backend.go

```diff
@@ -7,29 +7,42 @@ import (
 	"time"
 )
 
+type BackendType interface {
+	any | []byte
+}
+
 type Backend interface {
+	Name() string
+	Options() StoreOptions
+}
+
+type BackendAny interface {
+	Backend
 	Store[any]
 }
 
+type BackendBytes interface {
+	Backend
+	Store[[]byte]
+}
```

Comment:
I'm not a big fan of these two new interfaces :) Are they necessary?

Reply:
I have changed for generic BackendTyped[T].